src/SamplesApp/UITests.Shared/Microsoft_UI_Xaml_Controls/RatingControlTests/RatingControlPage.xaml.cs

```diff
@@ -0,0 +1,639 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+using Windows.UI.Xaml.Automation;
+using Windows.UI.Composition;
+using Windows.UI.Xaml.Hosting;
+using System.Numerics;
+using Windows.UI;
+using Windows.Foundation.Metadata;
+using RatingControl = Microsoft.UI.Xaml.Controls.RatingControl;
+using RatingItemImageInfo = Microsoft.UI.Xaml.Controls.RatingItemImageInfo;
+using RatingItemFontInfo = Microsoft.UI.Xaml.Controls.RatingItemFontInfo;
+using Uno.UI.Samples.Controls;
+
+namespace UITests.Microsoft_UI_Xaml_Controls.RatingControlTests
+{
+	[Sample("RatingControl", "WinUI")]
+	public sealed partial class RatingControlPage : Page
+	{
+		UIElement _secondTextBlockUI;
+		DispatcherTimer _dt;
+
+		SolidColorBrush _tomato = new SolidColorBrush(Colors.Tomato);
+		SolidColorBrush _aqua = new SolidColorBrush(Colors.Aqua);
+
+		public RatingControlPage()
+		{
+			this.InitializeComponent();
+
+			TestRatingControl.ValueChanged += TestRatingControl_ValueChanged;
+
+			RatingDarkTheme.PlaceholderValue = 1.5;
+
+			MyRatingReadOnlyTextBlock.Text = "2.2";
+			MyRatingReadOnlyWithPlaceholder.PlaceholderValue = 3.3;
+
+			DisabledWithValue.Value = 3;
+			DisabledWithPlaceholderValue.PlaceholderValue = 3;
+
+			CustomImages.Value = 3.0;
+			CustomImages.PlaceholderValue = 1.5;
+
+			var imageInfo = new RatingItemImageInfo();
+			imageInfo.Image = new Windows.UI.Xaml.Media.Imaging.BitmapImage(new Uri("ms-appx:/Assets/rating_set.png"));
+			imageInfo.UnsetImage = new Windows.UI.Xaml.Media.Imaging.BitmapImage(new Uri("ms-appx:/Assets/rating_unset.png")); ;
+			imageInfo.PlaceholderImage = new Windows.UI.Xaml.Media.Imaging.BitmapImage(new Uri("ms-appx:/Assets/rating_placeholder.png")); ;
+			imageInfo.DisabledImage = new Windows.UI.Xaml.Media.Imaging.BitmapImage(new Uri("ms-appx:/Assets/rating_disabled.png")); ;
```

Comment:
Codacy found an issue: [Remove this empty statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6162540)